Packages/MovieDBUI/Sources/MovieDBUI/UI/Screens/MovieDetails/MovieDetailsViewController.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  MovieDetailsViewController.swift
+//  MovieDB
+//
+//  Created by Mouad Bj on 16/3/2023.
+//
+
+import UIKit
+import MovieDBCore
+
+public class MovieDetailsViewController: UIViewController, NibLoadable {
+
+  public var viewModel: MovieDetailsViewModel!
+  
+  @IBOutlet weak var poster: UIImageView!
+
+  @IBOutlet weak var titleLabel: UILabel!
+  @IBOutlet weak var genresLabel: UILabel!
+  @IBOutlet weak var descLabel: UILabel!
+  
+  @IBOutlet weak var mainView: UIView!
+  @IBOutlet weak var loadingIndicator: UIActivityIndicatorView!
+  
+  var bag = DisposeBag()
+  
+  public override func viewDidLoad() {
+    super.viewDidLoad()
+    setup()
+    viewModel.loadDetails()
+  }
+  
+  func setup() {
+    viewModel.$movieDetails
+      .map(\.value)
+      .replaceError(with: nil)
+      .compactMap { $0 }
+      .receive(on: DispatchQueue.main)
+      .sink(receiveValue: { [weak self] movieDetail in
+        if let posterURL = self?.viewModel.posterURL {
+          self?.poster.load(url: posterURL, placeholder: nil)
+        }
+        self?.titleLabel.text = movieDetail.title ?? "-"
```

Comment:
I Agree, I would say data adaptation should be in the ViewModel and not the ViewController. or in some cases even in the domain model? what do u say? However, the domain model should be application agnostic. so if the unwrapping will be in the domain layer it shouldn't be explicit to the app but more generic to the business itself rather than to the application.

Reply:
Yes that's what I meant by 
> or in some cases even in the domain model
- if the unwrapped value is related to the business logic then it should be in the domain layer (i.e. `.transform()` will do the unwrapping and place the actual unwrapped values in the domain models) [e.g. if the business chose that `Movie.title` should never be nil, then `.transform()` will unwrap the variable and place it into the domain model which accepts a non-nil title only]
- if the unwrapping/adaptation is application-related then it should be in presentation adapters (VM, Presenters...) [e.g. in some cases the same date field can be displayed differently in different screens so it's up to the presentation layer to adapt the Date field to the appropriate string format for every screen (Comment.publishDate, could be displayed [11/11/2011] in one screen, and [11 years ago] in another), but the domain model will probably only care if the Date field is nil or not]